source/common/http/http2/metadata_decoder.cc

```diff
@@ -54,22 +54,23 @@ bool MetadataDecoder::decodeMetadataPayloadUsingNghttp2(bool end_metadata) {
     // is_end indicates if the data in slice is the last data in the current
```

Comment:
The reason for the changes in the source/common/http/http2/metadata_decoder.cc is that, while iterating on this PR, I changed the `RawSlice` declaration to make the `mem_` field immutable by code outside the buffer implementation. The one place that modified that field was the metadata encoder, so I changed metadata_decoder.cc to not require the field to be mutable.

I later reverted the having `RawSlice.mem_` be mutable, so the metadata_decoder.cc change here isn't strictly necessary any more. I do like the idea of keeping this change, though, because there are cases where it's unsafe to modify that `mem_` field, and the change here removes an example that people might otherwise copy and unwittingly use in an unsafe context.

Reply:
I agree we should change this, but do you mind moving into a different PR?

Reply:
Ok, I've removed it from this PR.